Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -316,6 +428,14 @@ public Task<TResult> InvokeWithParameterObjectAsync<TResult>(string targetName,
                                     capabilities["workspace"] = JToken.FromObject(new { });
                                 }
                                 capabilities["workspace"]["workspaceFolders"] = true;
+                                capabilities["workspace"]["didChangeWatchedFiles"]["dynamicRegistration"] = true;
+
+                                // Root path and root URI should not be sent. They're deprecated and will
+                                // just confuse pylance with respect to what is the root folder. 
+                                // https://microsoft.github.io/language-server-protocol/specifications/specification-current/#initialize
+                                // Setting them to empty will make pylance think they're gone.
+                                messageParams["rootPath"] = "";
```

Comment:
New integration test found this bug. rootPath being set forces Pylance to pick this as a workspace folder. Then later when the same folder is sent for workspaceFolderChanged, Pylance ends up with two workspaces with the same path. This messes up its internal map and makes it look like the workspace never finishes loading.

These two properties are deprecated and replaced by the workspace folder notification in the LSP so I just removed them.

Reply:
hmm. we might need the project root when searching for a pyright configfile.

eric just fixed a bug related to this. https://github.com/microsoft/pyright/commit/e25b2c71df40f6ef195f77c03d79a4b0f26a6495

Reply:
workspaceFolderChanged should cause the root to be set.